### ISSUE TITLE
Crash on startup on RPI3 on RetroPie

### ISSUE THREAD

**Issue by oooMaNiXooo** [BlitterStudio/amiberry#1179]
**Describe the bug**
Hi Guys, I hope this is an acceptable place to log this issue.  I am having issues running Amiberrry v5.6.3 on a RPI3 using a clean 4.8.5 (Buster) image from the Retropie website.  I have flashed the image a few times, installed Amiberry from binary and also source but am unable to get it to run.  All kickstarts are correct with MD5 checked, deployment guide followed to the letter.  When I launch a game (any game) (WHDLoad) it hangs on a black screen with a cursor in the top left corner.  If I try to launch ./amiberry from command line, the GUI opens then immediately closes.  I have tried everything from cleaning out .cfg files to imaging fresh over and over, but the same issues persist.  I have an older build on a different SD card (stretch based), which works perfectly on the exact same hardware and controllers, but the latest build will not run.

**To Reproduce**

This is easily reproduced by writing a fresh image, installing Amiberry then trying to run it, same issue everytime.  I have tried using the same kickstart roms as my old build and also refreshed with new ones, all MD5 checked.  I tried installing [r-puae2021] using the same kickstart and games rims on the same hardware and it works fine.

**Expected behavior**
I would expect the emulator to function normally aligned with documentation and previous experience of using the emulator

**Screenshots**

**Desktop (please complete the following information):**

- OS: [e.g. Debian]
- 4.8.5 (Buster)

- Version [e.g. 5.4]
- 4.8.5 (Buster)
- Amiberrry v5.6.3

**Additional context**
While not an expert on linux I have been using Retropie and Amiberry for a long time, this is a new issue for me and I am uncertain why it is happening.  I have tried disconnecting the controllers from the pi, no impact.  I am using a ControlBlock to operate the controllers so that driver is installed and running

Love Amiberry? Please consider supporting it:
👉  https://ko-fi.com/midwan

**Comment by giantclambake** [comment]
Query: which rpi3 model?

**Comment by oooMaNiXooo** [comment]
Sorry, its Raspberry Pi 3 Model B Rev 1.2

**Comment by solskogen** [comment]
We just had someone else having the same problem. Could you try
disconnecting your controller? If that works, the problem is already solved
in the source.

**Comment by oooMaNiXooo** [comment]
Thanks :)  I have removed the two controllers so only the keyboard remains, but the same issue.  Do I need to remove the ControlBlock driver and joystick selection modules also?  Should I try to update from source?

**Comment by midwan** [comment]
@oooMaNiXooo Please try cloning this repo and compiling the latest sources, to see if that fixes the problem.
There was a bugfix just last night about something similar, perhaps it covers your case as well.

**Comment by oooMaNiXooo** [comment]
no problem i will do that thank you.  is it best to use compile from source within the RetroPie setup menu or via command line at console?

**Comment by midwan** [comment]
In this case, grab the sources manually. I believe RetroPie uses a fixed release tag when doing "compile from source", and if that's still true, you wouldn't get the latest ones from last night.

**Comment by oooMaNiXooo** [comment]
ah ok that's great thanks.  Would you mind dropping in a few instructions to ensure i do it correctly and don't waste anyones time? :)

**Comment by midwan** [comment]
Check the Readme in this repo, it should cover what you need :)

**Comment by oooMaNiXooo** [comment]
Thanks so I have managed to compile from source and have tested again, but results remain the same.  Launching a game leaves the screen black with a cursor.  launching ./amiberry flashes the same black screen with cursor then it closes.  run command.log shows only the following bash: /opt/retropie/supplementary/joystick-selection/js-onstart.sh: No such file or directory
Parameters:
Executing: /opt/retropie/emulators/amiberry/amiberry.sh "/home/pi/RetroPie/roms/amiga/StuntCarRacer_v1.3_0222.lha"
Launching ...
./amiberry --autoload /home/pi/RetroPie/roms/amiga/StuntCarRacer_v1.3_0222.lha -G

**Comment by midwan** [comment]
@oooMaNiXooo 
Then it's something else going on...
I tested both the RetroPie-supplied DMX version, as well as my own compile of the latest SDL2 version yesterday, and both worked fine for me. We'll have to figure out what is causing the issue in your device.

Can you try running Amiberry from the console, by exiting EmulationStation (with F4), navigating to `/opt/retropie/emulators/amiberry/` and launching it with `./amiberry` ?

Do you get to the GUI when you do that? (check if you have any config named `default.uae` - that would be auto-loaded and if it's specifying to hide the GUI and boot directly into emulation, that's what will happen).

If you do, what happens if you select A500 from Quickstart and press Start? Do you get to the Kickstart 1.3 screen?
If you don't and you're thrown back to the console, is there an error message shown?

Also, enable Amiberry logging by going in the Paths panel, and enabling that checkbox. It will try to write a logfile in the current directory the binary is started from, so you will need permissions to do that. An easy way is to run it with `sudo` first, another way is to manually change the permissions in that directory, yet another way is to change where the logfile is saved.

Please try to trigger the problem when logging is enabled, and post that logfile here, along with any error messages in the console (if any).

**Comment by oooMaNiXooo** [comment]
When i execute it from the command prompt as detailed above I see the same black screen flashing up with the cursor, then it goes off and returns me to the command prompt.  I am unable to see the GUI at all.  I did see the GUI yesterday sometimes when testing but it flashed on for a second then immediately went off again.  I have no way of enabling any additional logging.  Is it worth me deleting the BIOS ROMs to see if that allows the GUI to open is are the rooms not required for that to happen?

**Comment by oooMaNiXooo** [comment]
also i don't have a default.uae file, just one for a500 and one for a1200 but I am not using those when trying to open amiberry

**Comment by giantclambake** [comment]
You can edit the amiberry.conf file and set `write_logfile=yes`  ;)

**Comment by oooMaNiXooo** [comment]
where should the amiberry.conf file be located (path)?

**Comment by oooMaNiXooo** [comment]
i suspect it should be here but i do not appear to have one>  could that be the issue?
pi@retropie:/opt/retropie/emulators/amiberry/conf $ ls -l
total 48
-rw------- 1 pi   pi     518 Nov 15 16:20 joystick-selection.cfg
lrwxrwxrwx 1 root root    39 Nov 15 22:27 retroarch.cfg -> /opt/retropie/configs/all/retroarch.cfg
-rw------- 1 pi   pi   33760 Nov 15 22:07 retroarch.cfg.old
-rw------- 1 pi   pi     198 Nov 15 10:48 rp-a1200.uae
-rw------- 1 pi   pi     172 Nov 15 10:48 rp-a500.uae

**Comment by midwan** [comment]
If the file is not there already, it's not a problem - it's normally regenerated by Amiberry itself, if not found.
However, I find it strange that you don't have the default one there at all. Did you use RetroPie Setup to install Amiberry, or did you do it yourself?

Which PLATFORM option did you use to compile Amiberry from source? There are 2 that can run in RetroPie for your platform, the DMX version and the SDL2 one. It's worth testing the other one to see if it changes anything.

The BIOS files are not related to this issue, so you can leave those alone. Running Amiberry from the console, with no parameters, and with no `default.uae` file in conf, should bring up the GUI anyway.

**Comment by oooMaNiXooo** [comment]
ah ok.  So yes I installed from the GUI using the package included in the bistro.  I then updated online and also tried to install from source.  When I compiled the package earlier I just ran "make PLATFORM=rpi3".  Sorry if I missed some additional syntax.  I am happy to compile again if you can confirm anything I may have missed.  This is SO strange as I have built many images in the past with amiberry and never had a problem.

**Comment by midwan** [comment]
When I tested RetroPie yesterday (fresh install with the latest version available), it installed Amiberry 5.6.2 from binary. Does that work for you? I didn't see any issues here, though I tested this on an RPI4, not a 3 (which normally _shouldn't_ matter).

Then I compiled the SDL2 version to test that as well, which in your case would be `PLATFORM=rpi3-sdl2`. It's best if you do a `make clean` first, to get rid of the previously compiled objects, since you'll be compiling for a different target. Or just clone it in a different directory, so you can have both of them around, if you want.

Test if the SDL2 version works any better, it should at least get you to the GUI. Once we've established that, we can investigate further.

**Comment by oooMaNiXooo** [comment]
great so I am compiling an SDL2 version at the moment, will see how that goes.  As an aside, I just did a wget to grab 5.6.2 for rpi3 and have tried to run that.  It runs to the GUI, but then immediately closes.  Is it worth me trying to manually create a config file so I can enable logging?  the GUI remains on screen for about 1 second then returns to the command prompt

**Comment by midwan** [comment]
Sure, you can get the default amiberry.conf from this repo, put it in your conf folder, and it will get used. Just remember to edit it and set logging to enabled, before you run it. And make sure it has permissions to create the logfile, otherwise you won't get anything :)

**Comment by oooMaNiXooo** [comment]
so based upon 5.6.2 which i downloaded, i managed to get a log file, pasted below.  I see an error in there, does that help at all?

Amiberry v5.6.2 (2023-11-09) Logfile

Compiled against SDL2 v2.0.9, Linked against SDL2 v2.0.10
Allocated 16 MB for 24-bit area (0x20000000) and 640 MB for Z3 and RTG at fake address (0x30000000 - 0x58000020)
Sorting devices and modes...
0: 1920x1080, 32-bit (60)
1 display modes.
Desktop: W=1920 H=1080 B=32 HZ=60. CXVS=1920 CYVS=1080
Enumerating SDL2 playback devices...
Detected 2 sound playback devices
Sound playback device 0: bcm2835 HDMI 1, bcm2835 HDMI 1
Sound playback device 1: bcm2835 Headphones, bcm2835 Headphones
Enumerating SDL2 recording devices...
Detected 0 sound recording devices
Enumeration end
0:SDL2: bcm2835 HDMI 1, bcm2835 HDMI 1
1:SDL2: bcm2835 Headphones, bcm2835 Headphones
Enumerating recording devices:
done
clipboard_init
Joystick #0: ControlBlock SNES Gamepad
      GUID: 03000000010000000100000004000000
      Axes: 2
      Buttons: 8
      Balls: 0
Joystick #0 does not have a mapping available
Joystick name: 'ControlBlock SNES Gamepad', sanitized to: 'ControlBlock SNES Gamepad'
No Retroarch controller cfg file found, checking for mapping in retroarch.cfg
No Retroarch controller cfg file found, using the default mapping
Joystick #1: ControlBlock SNES Gamepad
      GUID: 03000000010000000100000004000000
      Axes: 2
      Buttons: 8
      Balls: 0
Joystick #1 does not have a mapping available
Joystick name: 'ControlBlock SNES Gamepad', sanitized to: 'ControlBlock SNES Gamepad'
No Retroarch controller cfg file found, checking for mapping in retroarch.cfg
No Retroarch controller cfg file found, using the default mapping
Joystick #2: ControlBlock Arcade Gamepad
      GUID: 03000000010000000100000004000000
      Axes: 2
      Buttons: 12
      Balls: 0
Joystick #2 does not have a mapping available
Joystick name: 'ControlBlock Arcade Gamepad', sanitized to: 'ControlBlock Arcade Gamepad'
No Retroarch controller cfg file found, checking for mapping in retroarch.cfg
No Retroarch controller cfg file found, using the default mapping
Joystick #3: ControlBlock Arcade Gamepad
      GUID: 03000000010000000100000004000000
      Axes: 2
      Buttons: 12
      Balls: 0
Joystick #3 does not have a mapping available
Joystick name: 'ControlBlock Arcade Gamepad', sanitized to: 'ControlBlock Arcade Gamepad'
No Retroarch controller cfg file found, checking for mapping in retroarch.cfg
No Retroarch controller cfg file found, using the default mapping
KS ver = 0 (0x00)
Stored port 0/0 d=0: added 200 0  
Stored port 0/0 d=0: added 200 0 System mouse MOUSE0
Port0: ID 'mouse' matched
Stored port 1/0 d=0: added 100 0  
Stored port 1/0 d=0: added 100 0 ControlBlock SNES Gamepad JOY0
Port1: ID 'joy0' matched
Port2: NONE
Port3: NONE
target_cfgfile_load(): load file /home/pi/amiberry5.6.2/conf/default.uae
load config '/home/pi/amiberry5.6.2/conf/default.uae':3
cfgfile_load_2 failed
KS ver = 0 (0x00)
Stored port 0/0 d=0: added 200 0  
Stored port 0/0 d=0: added 200 0 System mouse MOUSE0
Port0: ID 'mouse' matched
Stored port 1/0 d=0: added 100 0  
Stored port 1/0 d=0: added 100 0 ControlBlock SNES Gamepad JOY0
Port1: ID 'joy0' matched
Port2: NONE
Port3: NONE
failed to load config '/home/pi/amiberry5.6.2/conf/default.uae'
KS ver = 0 (0x00)
Stored port 0/0 d=1: added 200 0 System mouse MOUSE0
Port0: COMBO 'System mouse' + 'MOUSE0' matched
Stored port 1/0 d=1: added 100 0 ControlBlock SNES Gamepad JOY0
Port1: COMBO 'ControlBlock SNES Gamepad' + 'JOY0' matched
Port2: NONE
Port3: NONE
Autoconfig board list:
Card 01: 'UAE Boot ROM' no autoconfig 00f00000 - 00f0ffff.
Card 02: 'UAE FS ROM'
  d1.52.40.00.07.db.00.00.00.03.20.00.00.00.00.00
  MID 2011 (07db) PID 82 (52) SER 00000003
  Z2 0x00e90000 0x00e90000   64K ROM 0
END
Autoconfig board list:
END
Stored port 0/1 d=0: added 200 0 System mouse MOUSE0
Stored port 1/1 d=0: added 100 0 ControlBlock SNES Gamepad JOY0
GUI: SDL2 Controller/Joystick added or removed, re-running import joysticks...
Joystick #0: ControlBlock SNES Gamepad
      GUID: 03000000010000000100000004000000
      Axes: 2
      Buttons: 8
      Balls: 0
Joystick #0 does not have a mapping available
--- New exception ---
JIT not in use.
Segmentation Fault
info.si_signo = 11
info.si_errno = 0
info.si_code = 1
info.si_addr = 00000000
r0  = 0x00000000
r1  = 0x7eb72e48
r2  = 0x00000008
r3  = 0x00000000
r4  = 0x7eb72e40
r5  = 0x00000000
r6  = 0x7eb72e38
r7  = 0x04230ea8
r8  = 0x7eb70dfc
r9  = 0x00000000
r10 = 0x0422fdb8
FP  = 0x00000000
IP  = 0x7eb72e40
SP  = 0x7eb70c28
LR  = 0x00000000
PC  = 0x76f81f94
CPSR = 0x68000010
Fault Address = 0x00000000
Trap no = 0x0000000e
Err Code = 0x00000805
Old Mask = 0x00000000
LR - 0x00000000: symbol not found
Stack trace:
0x0067ea68 <(null) + 0x0067ea68> (./amiberry)
0x769381c0 <__default_rt_sa_restorer + 0x00000000> (/lib/arm-linux-gnueabihf/libc.so.6)
01: 0x0422fdb8 <(null) + 0x0422fdb8> (./amiberry)
Stack trace (non-dedicated):
./amiberry(+0x27db18) [0x67eb18]
/lib/arm-linux-gnueabihf/libc.so.6(__default_rt_sa_restorer+0) [0x769381c0]
End of stack trace.
--- end exception ---

**Comment by midwan** [comment]
Yeah, that's exactly the bug we fixed yesterday night, so that should not occur with the latest version from `master` anymore. It was triggered in the initialization of some joystick controllers.

**Comment by oooMaNiXooo** [comment]
ah ok that's excellent news then.  i will download a copy of master and will test again :)

**Comment by oooMaNiXooo** [comment]
I am sure this is me, but I still have the same issue and error.  I fully expect this is me cloning something wrong, however the files i have cloned locally appear to have your fixes in, but the compiled version seems to function the same and produces the same error in the log.  i created a local folder, then ran  sudo git clone https://github.com/BlitterStudio/amiberry, then compiled using make and then ran the file using ./amiberry.  The result is the same black screen and cursor and same error in the log.  Did I get something wrong here?  So sorry to ask and I really appreciate your help on this! :)

**Comment by midwan** [comment]
And does it work if you disconnect all the joysticks and controllers (even adapters)?

**Comment by oooMaNiXooo** [comment]
only thing connected is my keyboard, which is am actual amiga 600 keyboard connected to USB via an adapter board.  Same hardware as used for the older version of amiberry on my other SD card without issues

**Comment by oooMaNiXooo** [comment]
ControlBlock also connected to the pi, but no controllers hooked in to the ports

**Comment by midwan** [comment]
But then it can't be the same log output, as the one above mentions: `Joystick #0: ControlBlock SNES Gamepad`, right? Can you try disconnecting that?

**Comment by cmitu** [comment]
It's probably fixed in #1178, ControlBlock will create the input devices even if there's nothing connected. Stop the controlblock service instead.

**Comment by oooMaNiXooo** [comment]
i think its picking that up as i have joystick selection installed which passes default controllers to amiberry.  i can uninstall that module?  there are no controllers plugged in to the pi

**Comment by oooMaNiXooo** [comment]
however ControlBlock drivers are running so i assume they are advertising the controllers?

**Comment by oooMaNiXooo** [comment]
still seems to be picking up the controllers even with the service stopped.  i will remove the module and try again :)

**Comment by cmitu** [comment]
Hm, it looks like the same crash - as in the issue reported yesterday - happens even without a gamepad connected and using current `master`. Log file (just running `amiberry` from the install folder with just the logging lines in `amiberry.conf`):

<details>
  <summary><i>amiberry.log</i></summary>

```
Amiberry v5.6.4 (2023-11-15) Logfile

Compiled against SDL2 v2.0.10, Linked against SDL2 v2.0.10
Allocated 16 MB for 24-bit area (0x20000000) and 640 MB for Z3 and RTG at fake address (0x30000000 - 0x58000020)
Sorting devices and modes...
0: 640x400, 8-bit (50,60)
1: 640x400, 16-bit (50,60)
2: 640x400, 32-bit (50,60)
3: 640x480, 8-bit (50,60)
4: 640x480, 16-bit (50,60)
5: 640x480, 32-bit (50,60)
6: 720x400, 8-bit (50,60)
7: 720x400, 16-bit (50,60)
8: 720x400, 32-bit (50,60)
9: 800x480, 8-bit (50,60)
10: 800x480, 16-bit (50,60)
11: 800x480, 32-bit (50,60)
12: 800x600, 8-bit (50,60)
13: 800x600, 16-bit (50,60)
14: 800x600, 32-bit (50,60)
15: 960x540, 8-bit (50,60)
16: 960x540, 16-bit (50,60)
17: 960x540, 32-bit (50,60)
18: 1024x768, 8-bit (50,60)
19: 1024x768, 16-bit (50,60)
20: 1024x768, 32-bit (50,60)
21: 1280x720, 8-bit (50,60)
22: 1280x720, 16-bit (50,60)
23: 1280x720, 32-bit (50,60)
24: 1280x800, 8-bit (50,60)
25: 1280x800, 16-bit (50,60)
26: 1280x800, 32-bit (50,60)
27: 1280x1024, 8-bit (50,60)
28: 1280x1024, 16-bit (50,60)
29: 1280x1024, 32-bit (50,60)
30: 1360x768, 8-bit (50,60)
31: 1360x768, 16-bit (50,60)
32: 1360x768, 32-bit (50,60)
33: 1366x768, 8-bit (50,60)
34: 1366x768, 16-bit (50,60)
35: 1366x768, 32-bit (50,60)
36: 1680x1050, 8-bit (50,60)
37: 1680x1050, 16-bit (50,60)
38: 1680x1050, 32-bit (50,60)
39: 1920x1080, 8-bit (50,60)
40: 1920x1080, 16-bit (50,60)
41: 1920x1080, 32-bit (50,60)
42 display modes.
Desktop: W=800 H=600 B=24 HZ=0. CXVS=800 CYVS=600
Enumerating SDL2 playback devices...
Detected 2 sound playback devices
Sound playback device 0: bcm2835 HDMI 1, bcm2835 HDMI 1
Sound playback device 1: bcm2835 Headphones, bcm2835 Headphones
Enumerating SDL2 recording devices...
Detected 0 sound recording devices
Enumeration end
0:SDL2: bcm2835 HDMI 1, bcm2835 HDMI 1
1:SDL2: bcm2835 Headphones, bcm2835 Headphones
Enumerating recording devices:
done
clipboard_init
Controller init_kb_from_retroarch(0): Found 
Controller init_kb_from_retroarch(1): Not found 
KS ver = 33 (0x21)
Stored port 0/0 d=0: added 200 0  
Stored port 0/0 d=0: added 200 0 System mouse MOUSE0
Port0: ID 'mouse' matched
Port1: ID 'joy0' matched
Port2: NONE
Port3: NONE
target_cfgfile_load(): load file /opt/retropie/emulators/amiberry/conf/default.uae
load config '/opt/retropie/emulators/amiberry/conf/default.uae':3
cfgfile_load_2 failed
KS ver = 33 (0x21)
Stored port 0/0 d=0: added 200 0  
Stored port 0/0 d=0: added 200 0 System mouse MOUSE0
Port0: ID 'mouse' matched
Port1: ID 'joy0' matched
Port2: NONE
Port3: NONE
failed to load config '/opt/retropie/emulators/amiberry/conf/default.uae'
KS ver = 33 (0x21)
Stored port 0/0 d=1: added 200 0 System mouse MOUSE0
Port0: COMBO 'System mouse' + 'MOUSE0' matched
Port1: ID 'joy0' matched
Port2: NONE
Port3: NONE
Stored port 0/1 d=0: added 200 0 System mouse MOUSE0
Autoconfig board list:
Card 01: 'UAE Boot ROM' no autoconfig 00f00000 - 00f0ffff.
Card 02: 'Pre-KS 1.3 UAE FS ROM'
  c1.52.40.00.07.db.00.00.00.03.00.00.00.00.00.00
  MID 2011 (07db) PID 82 (52) SER 00000003
  Z2 0x00e90000 0x00e90000   64K IO  0
END
Autoconfig board list:
END
--- New exception ---
JIT not in use.
Segmentation Fault
info.si_signo = 11
info.si_errno = 0
info.si_code = 1
info.si_addr = 00000000
r0  = 0x76f74480
r1  = 0x00795298
r2  = 0x0000001c
r3  = 0x769dbc70
r4  = 0x03f2333c
r5  = 0x033609c0
r6  = 0x034ce9c0
r7  = 0x05a95aa0
r8  = 0x0073eb8c
r9  = 0x00748b40
r10 = 0x05a95960
FP  = 0x0000000a
IP  = 0xffffffff
SP  = 0x7ec5f1ac
LR  = 0x769acdc8
PC  = 0x7690b4f4
CPSR = 0x80000010
Fault Address = 0x00000000
Trap no = 0x0000000e
Err Code = 0x00000017
Old Mask = 0x00000000
LR - 0x769ACDC8: <(null)> (/lib/arm-linux-gnueabihf/libc.so.6)
Stack trace:
0x0021cee4 <(null) + 0x0021cee4> (./amiberry)
0x768bf1c0 <__default_rt_sa_restorer + 0x00000000> (/lib/arm-linux-gnueabihf/libc.so.6)
IP out of range
Stack trace (non-dedicated):
./amiberry() [0x21d004]
/lib/arm-linux-gnueabihf/libc.so.6(__default_rt_sa_restorer+0) [0x768bf1c0]
End of stack trace.
--- end exception ---
```
</details>
I'll try to see if the same thing happens with 5.6.1, which has been the previous version RetroPie had in place.

**Comment by oooMaNiXooo** [comment]
Further update from me guys.  I have removed the ControlBlock package and joystick selection package, but still the same crash.  I suspect that it knows the ControlBlock is there even without the software installed.  it could be some config files have been left behind. all i could do next is a fresh image again, but not install anything but amiberry then test.  is it worth it or is this a code issue that needs to be fixed?

**Comment by midwan** [comment]
@oooMaNiXooo Hang on, because I managed to recreate this... but only on the RPI3. The RPI4 seems to work normally with RetroPie's deployed version (5.6.3), but the RPI3 fails in the way you described above.

I'll try to debug this locally and figure out what's causing it, but now that I can recreate it I'm sure we'll find a solution.

**Comment by oooMaNiXooo** [comment]
ah that's great news and I am so pleased you are able to see the issue!...I will leave it with you and thanks so much again for being so helpful! :)

**Comment by midwan** [comment]
After a few more tests, I can see that it's only the DMX version that is failing to startup. The SDL2 one does work, without issues.
This may be an issue with the latest system updates then, if those broke dispmanx - but I'm not 100% sure. Does an older version work on the same system? If we find the last version that works, maybe we can track any changes, but there haven't been any related to gfx (and specifically DMX) lately.

You could use the sdl2 version of course, but I'm guessing it will be much slower and especially on the RPI3, it might be _too_ slow in some cases.

**Comment by cmitu** [comment]
I'll try to bisect this. The OS hasn't had much updates lately (i.e. 1 year at least) and the DipmanX related libs have certainly not been updated in a long time.

**Comment by midwan** [comment]
@cmitu Thanks, I'll also try to test older versions and see if it starts to break at some point.

**Comment by midwan** [comment]
I found the exact commit that broke it, and it's a silly change in the GUI. Fix incoming shortly...

**Comment by oooMaNiXooo** [comment]
I am also glad I didn't waste your time with a user error and that its a genuine bug

**Comment by midwan** [comment]
There was a Checkbox in the GUI that was added recently, and that would try to get the SDL2 back-end used to determine if it should be enabled or not. But under DispmanX, there is _not_ SDL2 back-end used, so that would trigger a segfault as soon as the GUI was rendered.

**Comment by oooMaNiXooo** [comment]
That's great mate, I will give it a test and report back but sounds like you nailed it! :)

**Comment by cmitu** [comment]
Thank you @midwan for the fix !
I can confirm that it works. I didn't finish my bisection, but compiled the latest master and the crash is gone.

Any chance for a new version number soon, to include this fix ?

**Comment by oooMaNiXooo** [comment]
i can confirm the crash has indeed gone with a build from the latest source, but I am unable to get anything WHDLoad related to run, i just get a ROM error.  I have imaged fresh again but so far no joy, however I will test further to make sure it isn't incompetence my side  Have you tested this running WHDLoad titles on RPI3 with the latest build?  If so please do let me know.  I will report back once I ave tested further.

**Comment by cmitu** [comment]
> If so please do let me know.

No, I haven't, I usually boot the Amiga test kit and test various things to make sure inputs are working, rendering is fine, etc. For this particular issue, just booting to the menu was fine, since it was crashing before without showing anything.

**Comment by oooMaNiXooo** [comment]
hey guys.  so I have tested things fully now and i think there is an issue running WHDLoad games with the latest version of amiberry.  I compiled an sdl2 version of 5.6.4 and the dispmanx version.  both open the GUI and can also get to the kickstart screen, but they will not load WHDLoad games.  I installed an old version 3.3 and it works fine.  if I change the amiberry file to either of the other two compiled versions is fails to run and hangs or displays a processor error.  I have a log for each build i have tested which are attached. the amiberry.log.panza was using the dispmanx build, the amiverry.log.stunt was using the sdl2 build.  I have used the same kickstarts etc in all cases.  is there anything in the logs which would suggest why its not working?
[amiberry.log.stunt.log](https://github.com/BlitterStudio/amiberry/files/13401787/amiberry.log.stunt.log)
[amiberry.log.panza.log](https://github.com/BlitterStudio/amiberry/files/13401788/amiberry.log.panza.log)
